tests/e2e/device-plugin_test.go

```diff
@@ -27,168 +27,154 @@ import (
 )
 
 // Actual test suite
-var _ = NVDescribe("GPU Device Plugin", func() {
-	f := framework.NewFramework("k8s-device-plugin")
-
-	Context("When deploying k8s-device-plugin", Ordered, func() {
-		// helm-chart is required
-		if *HelmChart == "" {
-			Fail("No helm-chart for k8s-device-plugin specified")
+var _ = Describe("GPU Device Plugin", Ordered, Label("gpu", "e2e", "device-plugin"), func() {
+	// Init global suite vars vars
+	var (
+		helmReleaseName string
+		chartSpec       helm.ChartSpec
+
+		collectLogsFrom      []string
+		diagnosticsCollector *diagnostics.Diagnostic
+	)
+
+	defaultCollectorObjects := []string{
+		"pods",
+		"nodes",
+		"namespaces",
+		"deployments",
+		"daemonsets",
+		"jobs",
+	}
+
+	values := helmValues.Options{
+		Values: []string{
+			fmt.Sprintf("image.repository=%s", ImageRepo),
+			fmt.Sprintf("image.tag=%s", ImageTag),
+			fmt.Sprintf("image.pullPolicy=%s", ImagePullPolicy),
+			"devicePlugin.enabled=true",
+			// We need to make affinity is none if not deploying NFD/GFD
+			// test will fail if not run on a GPU node
+			"affinity=",
+		},
+	}
+
+	// check Collector objects
+	collectLogsFrom = defaultCollectorObjects
+	if CollectLogsFrom != "" && CollectLogsFrom != "default" {
+		collectLogsFrom = strings.Split(CollectLogsFrom, ",")
+	}
+
+	BeforeAll(func(ctx SpecContext) {
+		// Create clients for apiextensions and our CRD api
+		helmReleaseName = "nvdp-e2e-test-" + randomSuffix()
+
+		chartSpec = helm.ChartSpec{
+			ReleaseName:   helmReleaseName,
+			ChartName:     HelmChart,
+			Namespace:     testNamespace.Name,
+			Wait:          true,
+			Timeout:       1 * time.Minute,
+			ValuesOptions: values,
+			CleanupOnFail: true,
 		}
 
-		// Init global suite vars vars
-		var (
-			crds      []*apiextensionsv1.CustomResourceDefinition
-			extClient *extclient.Clientset
-
-			helmReleaseName string
-			chartSpec       helm.ChartSpec
-
-			collectLogsFrom      []string
-			diagnosticsCollector *diagnostics.Diagnostic
-		)
-
-		defaultCollectorObjects := []string{
-			"pods",
-			"nodes",
-			"namespaces",
-			"deployments",
-			"daemonsets",
-			"jobs",
-		}
+		By("Installing k8s-device-plugin Helm chart")
+		_, err := helmClient.InstallChart(ctx, &chartSpec, nil)
+		Expect(err).NotTo(HaveOccurred())
 
-		values := helmValues.Options{
-			Values: []string{
-				fmt.Sprintf("image.repository=%s", *ImageRepo),
-				fmt.Sprintf("image.tag=%s", *ImageTag),
-				fmt.Sprintf("image.pullPolicy=%s", *ImagePullPolicy),
-				"devicePlugin.enabled=true",
-				// We need to make affinity is none if not deploying NFD/GFD
-				// test will fail if not run on a GPU node
-				"affinity=",
-			},
-		}
+		// Wait for all DaemonSets to be ready
+		// Note: DaemonSet names are dynamically generated with the Helm release prefix,
+		// so we wait for all DaemonSets in the namespace rather than specific names
+		By("Waiting for all DaemonSets to be ready")
+		err = internal.WaitForAllDaemonSetsReady(ctx, clientSet, testNamespace.Name)
```

Comment:
Do the DaemonSets also have app labels, or some other identifier that we could use to further restrict this set?